playwright/src/test/java/com/microsoft/playwright/Server.java

```diff
@@ -187,7 +193,19 @@ public void handle(HttpExchange exchange) throws IOException {
       path = "/index.html";
     }
 
-    // Resources from "src/test/resources/" are copied to "resources/" directory in the jar.
+    // If static files directory is set, serve from filesystem first
+    if (staticFilesDirectory != null) {
```

Comment:
underneath we do something similar but via `getResourceAsStream` which got introduced [here](https://github.com/microsoft/playwright-java/pull/657/files#diff-e20e283060b38eabf5fb0339495c168fa6545597a11be0aa13adc3760433514aR196) - do you remember why?

Reply:
This is the standard way to read resources in a maven package (they move from the source into the test package artifact). I'd keep using it with the custom base dir too, can we do that?

Reply:
I think there is no way of accessing the trace-viewer html source code via the tests/resource directory so I introduced another resourceLoader concept - are you fine with that?

Reply:
Yes. I'd pass the resource provide in server the constructor.